pkg/virt-handler/node-labeller/arch_labeller.go

```diff
@@ -0,0 +1,77 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright the KubeVirt Authors.
+ *
+ */
+
+package nodelabeller
+
+const (
+	amd64 = "amd64"
+	arm64 = "arm64"
+	s390x = "s390x"
+)
+
+// Ensure that there is a compile error should the struct not implement the archLabeller interface anymore.
+var _ = archLabeller(&defaultArchLabeller{})
+
+type archLabeller interface {
+	ShouldLabelNodes() bool
```

Comment:
Why do you mix exported/unexported methods in this interface?

Reply:
This one Method is needed outside of the package, the rest is only used internally.